tests/containers/bci_tests.pm

```diff
@@ -0,0 +1,135 @@
+    }
+
+    record_info('Install', 'Install needed packages');
+    zypper_call('--quiet in git-core python39', timeout => 600);
+    assert_script_run('pip3.9 --quiet install tox pytest', timeout => 600);
+
+    record_info('Clone', 'Clone BCI tests repository');
+    assert_script_run("git clone -q --depth 1 $bci_tests_repo");
+
+    record_info('Build', 'Build bci-tests project');
+    assert_script_run('cd bci-tests');
+    assert_script_run("export CONTAINER_RUNTIME=$runtime");
+    assert_script_run("export BCI_DEVEL_REPO=$bci_devel_repo") if $bci_devel_repo;
+    assert_script_run('tox -e build', timeout => 600);
+    upload_logs('junit_build_serial.xml');
+    upload_logs('junit_build_parallel.xml');
+
+    # bci-tests produce separate XUnit results files for each environment.
+    # We need tp merge all together into a single xml file that will
+    # be used by OpenQA to represent the results in "External results"
+    my $dom  = XML::LibXML::Document->new('1.0', 'utf-8');
+    my $root = $dom->createElement('testsuites');
+    $dom->setDocumentElement($root);
+
+    # Run the tests for each environment
+    for my $env (split(/,/, $test_envs)) {
+        record_info("Test: $env");
+        assert_script_run("tox -e $env", timeout => $bci_timeout);
+        upload_logs('junit_' . $env . '_serial.xml');
+        my $log_file = upload_logs('junit_' . $env . '_parallel.xml');
+        my $dom      = XML::LibXML->load_xml(location => "ulogs/$log_file");
+        for my $node ($dom->findnodes('//testsuite')) {
+            # Replace default attribute name "pytest" by its env name
+            $node->{name} =~ s/pytest/$env/;
+            # Append test results to the final xml file
+            $root->appendChild($node);
+        }
+    }
+    # Dump xml contents to a location where we can access later using data_url
+    $dom->toFile("os-autoinst-distri-opensuse/data/result.xml", 1);
+    # Download file from host pool to the instance
+    assert_script_run 'curl -f -v ' . data_url('result.xml') . " > /tmp/result.txt";
+    parse_extra_log('XUnit', '/tmp/result.txt');
+}
+
+sub test_flags {
+    return {fatal => 1, milestone => 1};
+}
+
+1;
+
+
+=head1 Discussion
+
+This module is used to test BCI repository within a container.
+
+It installs the required python packages and uses the existing BCI-test
+repository defined by B<BCI_TESTS_REPO>. Then, it will run the different
+tests environments defined by B<BCI_TEST_ENVS>.
+
+
+=head1 Configuration
+
+=head2 RUNTIME
+
+The runtime to be used by the tests. It can only be B<docker> or B<podman>.
+
+=head2 BCI_TESTS_REPO
+
+The location of the bci-tests repository to be cloned.
+
+=head2 BCI_TEST_ENVS
+
+The list of environments to be tested.
+e.g. base,init,dotnet,python,node,go,multistage
+
+=head2 BCI_DEVEL_REPO
+
+This parameter is given to the bci-tests to inject a different SLE_BCI
+repository url to the container image instead of the default one.
```

Comment:
as `my $bci_tests_repo = get_required_var('BCI_TESTS_REPO');` what should be the default repository? Should it be provided by some other tool as SCC?

Reply:
As sorry misread the variable name.

Reply:
Good point. For now, it's in gitlab, and in an individual's namespace. So, I'd keep it like this until we have a more official placement for the repo.

Reply:
And I misunderstood your question :D

Reply:
Mondays are hard :D